content/ember/v5/deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,1265 @@
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObject(value) {
+    this.abc = [...this.abc, value];
+  }
+}
+```
+
+#### `pushObjects`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObjects(values) {
+    this.abc.pushObjects(values);
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  pushObjects(values) {
+    this.abc.splice(this.abc.length, 0, ...values);
+  }
+}
+```
+or
+
+```js
+// @tracked
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObjects(values) {
+    this.abc = [...this.abc, ...values];
+  }
+}
+```
+
+#### `removeAt`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  removeAt(start, len) {
+    this.abc.removeAt(start, len);
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  removeAt(start, len) {
+    this.abc.splice(start, len);
+  }
+}
+```
+or
+
+```js
+// @tracked
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  removeAt(start, len) {
+    this.abc.splice(start, len);
+    this.abc = [...this.abc];
+  }
+}
+```
+
+#### `removeObject`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  removeObject(value) {
+    this.abc.removeObject(value);
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  removeObject(value) {
+    let loc = this.abc.length || 0;
+    while (--loc >= 0) {
+      let curValue = this.abc.at(loc);
+
+      if (curValue === value) {
+        this.abc.splice(loc, 1);
+      }
+    }
```

Comment:
Wouldn't a simple filer call work here?

```suggestion
    this.abc.filter(item => item !== value);
```

same for the related examples below...